x-pack/libbeat/autodiscover/providers/aws/elb/provider.go

```diff
@@ -0,0 +1,146 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package elb
+
+import (
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/aws/external"
+	"github.com/aws/aws-sdk-go-v2/service/elasticloadbalancingv2"
+	"github.com/aws/aws-sdk-go-v2/service/elasticloadbalancingv2/elasticloadbalancingv2iface"
+	"github.com/gofrs/uuid"
+
+	"github.com/elastic/beats/libbeat/autodiscover"
+	"github.com/elastic/beats/libbeat/autodiscover/template"
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/common/bus"
+	"github.com/elastic/beats/libbeat/common/cfgwarn"
+	"github.com/elastic/beats/libbeat/logp"
+)
+
+func init() {
+	autodiscover.Registry.AddProvider("aws_elb", AutodiscoverBuilder)
+}
+
+// Provider implements autodiscover provider for aws ELBs.
+type Provider struct {
+	fetcher       fetcher
+	period        time.Duration
+	config        *Config
+	bus           bus.Bus
+	builders      autodiscover.Builders
+	appenders     autodiscover.Appenders
+	templates     *template.Mapper
+	startListener bus.Listener
+	stopListener  bus.Listener
+	watcher       *watcher
+	uuid          uuid.UUID
+}
+
+// AutodiscoverBuilder is the main builder for this provider.
+func AutodiscoverBuilder(bus bus.Bus, uuid uuid.UUID, c *common.Config) (autodiscover.Provider, error) {
+	cfgwarn.Experimental("aws_elb autodiscover is experimental")
+
+	config := defaultConfig()
+	err := c.Unpack(&config)
+	if err != nil {
+		return nil, err
+	}
+
+	var clients []elasticloadbalancingv2iface.ClientAPI
+	for _, region := range config.Regions {
+		awsCfg, err := external.LoadDefaultAWSConfig()
```

Comment:
I think LoadDefaultAWSConfig can be outside this for loop because besides region is different for each `awsCfg`, the credentials will be the same.